OCPBUGS#29279: Doc improvements (#71446)
openshift/openshift-docs: modules/lvms-creating-volume-snapshots.adoc

// Module included in the following assemblies:
//
// storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc

:_mod-docs-content-type: PROCEDURE
[id="lvms-creating-volume-snapshots_{context}"]
= Creating volume snapshots

You can create volume snapshots based on the available capacity of the thin pool and the over-provisioning limits.
To create a volume snapshot, you must create a `VolumeSnapshotClass` custom resource (CR).

.Prerequisites

* You ensured that the persistent volume vlaim (PVC) is in `Bound` state. This is required for a consistent snapshot.
* You stopped all the I/O to the PVC.

.Procedure

. Log in to the OpenShift CLI (`oc`).
+
[IMPORTANT]
====
If you are using {lvms} in {rh-rhacm}, log in to the {rh-rhacm} CLI using your {product-title} credentials.
====

> . Create a `VolumeSnapshotClass` CR YAML file:

Reviewer: ```suggestion
. Create a `VolumeSnapshot` CR YAML file:
```

Author: Updated as per the suggestion.

+

> .Example `VolumeSnapshotClass` CR to create a volume snapshot

Reviewer: ```suggestion
.Example `VolumeSnapshot` CR to create a volume snapshot
```

[source,yaml]
----

> apiVersion: snapshot.storage.k8s.io/v1
> kind: VolumeSnapshot
> metadata:
>     name: lvm-block-1-snap <1>
> spec:
>     volumeSnapshotClassName: lvms-vg1 <2>
>     source:
>         persistentVolumeClaimName: lvm-block-1

Reviewer: ```suggestion
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshot
metadata:
  name: lvm-block-1-snap <1>
spec:
  volumeSnapshotClassName: lvms-vg1 <2>
  source:
    persistentVolumeClaimName: lvm-block-1
```

Author: Updated as per the suggestion.

----
<1> Specify a name for the volume snapshot.

> <2> Set this field to the value of the `storageClassName` field in the `PersistentVolumeClaim` CR of the source PVC.  

Reviewer: ```suggestion
<2> Set this field to the name of either default `volumeSnapshotClassName` present in cluster or custom `volumeSnapshotClassName` 
```

Author: @radeore 
Is this field ``volumeSnapshotClassName` nothing but the name of the volume snapshot?

Author: Updated as per the suggestion.

> . Create the `VolumeSnapshotClass` CR in the namespace where you created the source PVC by running the following command:

Reviewer: ```suggestion
. Create the `VolumeSnapshot` CR in the namespace where you created the source PVC by running the following command:
```

Author: Updated as per the suggestion.

+
[source,terminal]
----
# oc create -f <file_name> <namespace>
----

> Upon creating the `VolumeSnapshotClass` CR, {lvms} creates a read-only copy of the PVC as a volume snapshot.

Reviewer: ```suggestion
Upon creating the `VolumeSnapshot` CR, {lvms} creates a read-only copy of the PVC as a volume snapshot.
```

// Module included in the following assemblies:
//
// storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc

:_mod-docs-content-type: PROCEDURE
[id="lvms-creating-volume-snapshots_{context}"]
= Creating volume snapshots

You can create volume snapshots based on the available capacity of the thin pool and the over-provisioning limits.
To create a volume snapshot, you must create a `VolumeSnapshotClass` object.

.Prerequisites

* You have access to {product-title} as a user with `cluster-admin` permissions.

> * You ensured that the persistent volume vlaim (PVC) is in `Bound` state. This is required for a consistent snapshot.

Reviewer 2: Was just glancing, and wondering if this is a typo, and if it should say "persistent volume claim" instead of "persistent volume vlaim."

Author: Hi @gwynnemonahan 
Yes it should be "claim' instead of "vlaim". Thank you for pointing this out.
